src/Query/QueryLockParser.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\DBAL\Query;
+
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+use Doctrine\DBAL\Platforms\SQLServerPlatform;
+
+use function implode;
+use function ksort;
+use function ltrim;
+use function rtrim;
+use function trim;
+
+final class QueryLockParser
```

Comment:
Naming: This "parser" does not actually _parse_ anything does it?

Reply:
Well, it translates from an array of strings (the locks "names") into the appropriate SQL.
I don't like the name either, but couldn't find anything better.

Any suggestion?

Reply:
I renamed it to `QueryLockBuilder`